lib/keystores.js

```diff
@@ -156,16 +157,22 @@ api.update = async ({config} = {}) => {
 
   // insert the configuration and get the updated record
   const now = Date.now();
-
-  const result = await database.collections.kmsKeystore.update({
+  const set = {
+    config,
+    'meta.updated': now
+  };
+  const query = {
     id: database.hash(config.id),
+    controller: database.hash(config.controller),
     'config.sequence': config.sequence - 1
-  }, {
-    $set: {
-      config,
-      controller: database.hash(controller),
-      'meta.updated': now
-    }
+  };
+  //update controller only if allowControllerUpdate and flag is true
```

Comment:
style: single space after `//`

```suggestion
  // update controller only if allowControllerUpdate and flag is true
```